### PR TITLE
Several GI related optimizations and fixes

### PR DESCRIPTION
**Pull request by reduz**
-SDFGI direct light is done over many frames
-SDFGI Changed settings for rays/frame
-SDFGI Misc optimizations
-SDFGI Bug fix on probe scroll

-GIProbe was not working, got it to work again
-GIProbe dynamic objects were not working, fixed

-Added a half size GI option.

-Added a --profile-gpu command line option to make it easier to profile graphics from command line.

-All shadow rendering is done with raster now (no compute)
-All shadow rendering is done by rendering directly to the shadow atlas
-Improved how buffer clearing is done to optimize the above.
-16 bits shadow map support, and made the default

### CODE REVIEW

File: servers/rendering/renderer_rd/shaders/resolve.glsl [hunk 1]
```diff
@@ -58,6 +58,117 @@ void main() {
...
+	vec4 group3;
+	vec4 group4;
+	int best_index = 0;
+
+	//2X
+	group1.x = texelFetch(source_depth, pos, 0).r;
+	group1.y = texelFetch(source_depth, pos, 1).r;
+
+	//4X
+	if (params.sample_count <= 4) {
```
Comment (clayjohn) [inline servers/rendering/renderer_rd/shaders/resolve.glsl:74]: Shouldnt these be >= ?

Right now a sample count of 2 will run all the conditions, while a sample count of 16 only uses the last

File: servers/rendering/renderer_rd/shaders/resolve.glsl [hunk 2]
```diff
@@ -58,6 +58,117 @@ void main() {
...
+			best_index = 1;
+			min_f = freq.y;
+		}
+		if (freq.z < min_f) {
+			best_index = 2;
+			min_f = freq.z;
+		}
+		if (freq.w < min_f) {
+			best_index = 3;
+			min_f = freq.w;
```
Comment (JFonS) [inline servers/rendering/renderer_rd/shaders/resolve.glsl:118]: Probably compiled away, but `min_f` is not used any more here.

File: servers/rendering/renderer_rd/shaders/resolve.glsl (hunk 1 above)
Comment (JFonS) [reply servers/rendering/renderer_rd/shaders/resolve.glsl:74]: Agreed.

**Review by clayjohn — APPROVED** [review]

**Comment by reduz** [comment]
Do not merge yet

**Comment by reduz** [comment]
Ok now can merge if CI passes.

**Comment by mrjustaguy** [comment]
This PR Causes a Shadow Bug on Nvidia Quadro K2000 Driver 452.06, W10 64 bit

Vulkan Shadow Bug

[Shadow.zip](https://github.com/godotengine/godot/files/5866246/Shadow.zip)

Accurate Behavior Prior to This PR:

Vulkan Shadow Before

**Comment by akien-mga** [comment]
Thanks!

**Comment by JFonS** [comment]
@mrjustaguy  Looks like the shadow masks is not being taken into account, so the origin lines are casting a shadow.

**Comment by reduz** [comment]
Indeed it looks like masking code or the cast shadow property is not being respected.. but I did not touch anything related to that in this PR so my guess is that it happened in an earlier one.

**Comment by Xartorx** [comment]
@JFonS @reduz it's probably [this](https://github.com/godotengine/godot/issues/45109) bug, or somewhat related bug.

**Comment by mrjustaguy** [comment]
You're Right, however for some reason it only manifests in this PR for me (Haven't seen it on any others thus far).. I removed the sphere and it was still shadowing the Line.. Refreshing the shadows however seems to Fix it

**Pull request merged**